### ISSUE TITLE
Centos7: No package nvidia-docker2 available for docker-ce-18.06.2.ce-3.el7

### ISSUE THREAD

**Issue by soul0517** [NVIDIA/nvidia-docker#959]
_The template below is mostly useful for bug reports and support questions. Feel free to remove anything which doesn't apply to you and add more information where it makes sense._

_Also, before reporting a new issue, please make sure that:_

- _You read carefully the [documentation and frequently asked questions](https://github.com/NVIDIA/nvidia-docker/wiki)._
- _You [searched](https://github.com/NVIDIA/nvidia-docker/issues?utf8=%E2%9C%93&q=is%3Aissue) for a similar issue and this is not a duplicate of an existing one._
- _This issue is not related to [NGC](https://github.com/NVIDIA/nvidia-docker/wiki/NGC), otherwise, please use the [devtalk forums](https://devtalk.nvidia.com/default/board/200/nvidia-gpu-cloud-ngc-users/) instead._
- _You went through the [troubleshooting](https://github.com/NVIDIA/nvidia-docker/wiki/Troubleshooting) steps._

---

### 1. Issue or feature description
The nvidia-docker2 package cannot be installed when using docker-ce-18.06.2.ce-3.el7.

### 2. Steps to reproduce the issue
# yum search --showduplicates nvidia-docker2
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
========================================================================================== N/S matched: nvidia-docker2 ===========================================================================================
nvidia-docker2-2.0.0-1.docker1.12.6.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.0-1.docker17.03.2.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.0-1.docker17.06.1.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.0-1.docker17.06.2.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.0-1.docker17.09.0.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.1-1.docker1.12.6.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.1-1.docker1.13.1.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.1-1.docker17.03.2.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.1-1.docker17.06.2.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.1-1.docker17.09.0.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.1-1.docker17.09.1.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.2-1.docker1.12.6.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.2-1.docker1.13.1.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.2-1.docker17.03.2.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.2-1.docker17.06.2.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.2-1.docker17.09.0.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.2-1.docker17.09.1.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.2-1.docker17.12.0.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker1.12.6.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker1.13.1.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker17.03.2.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker17.06.2.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker17.09.0.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker17.09.1.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker17.12.0.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker17.12.0.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker17.12.1.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.03.0.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.03.1.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.06.0.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.06.1.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.06.1.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.06.2.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.09.0.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.09.1.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.09.2.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.09.2.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.09.3.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-1.docker18.09.4.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-2.docker18.06.2.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-2.docker18.09.5.ce.noarch : nvidia-docker CLI wrapper
nvidia-docker2-2.0.3-3.docker18.09.5.ce.noarch : nvidia-docker CLI wrapper

# yum install nvidia-docker2-2.0.3-1.docker18.06.2
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Resolving Dependencies
--> Running transaction check
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker18.06.2 will be installed
--> Processing Dependency: nvidia-container-runtime = 2.0.0-1.docker18.06.2 for package: nvidia-docker2-2.0.3-1.docker18.06.2.noarch
--> Processing Dependency: docker = 2:18.06.2 for package: nvidia-docker2-2.0.3-1.docker18.06.2.noarch
--> Running transaction check
---> Package nvidia-container-runtime.x86_64 0:2.0.0-1.docker18.06.2 will be installed
--> Processing Dependency: nvidia-container-runtime-hook < 2.0.0 for package: nvidia-container-runtime-2.0.0-1.docker18.06.2.x86_64
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker18.06.2 will be installed
--> Processing Dependency: docker = 2:18.06.2 for package: nvidia-docker2-2.0.3-1.docker18.06.2.noarch
--> Running transaction check
---> Package nvidia-container-runtime-hook.x86_64 0:1.4.0-2 will be installed
--> Processing Dependency: libnvidia-container-tools < 2.0.0 for package: nvidia-container-runtime-hook-1.4.0-2.x86_64
--> Processing Dependency: libnvidia-container-tools >= 0.1.0 for package: nvidia-container-runtime-hook-1.4.0-2.x86_64
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker18.06.2 will be installed
--> Processing Dependency: docker = 2:18.06.2 for package: nvidia-docker2-2.0.3-1.docker18.06.2.noarch
--> Running transaction check
---> Package libnvidia-container-tools.x86_64 0:1.0.2-1 will be installed
--> Processing Dependency: libnvidia-container1(x86-64) >= 1.0.2-1 for package: libnvidia-container-tools-1.0.2-1.x86_64
--> Processing Dependency: libnvidia-container.so.1(NVC_1.0)(64bit) for package: libnvidia-container-tools-1.0.2-1.x86_64
--> Processing Dependency: libnvidia-container.so.1()(64bit) for package: libnvidia-container-tools-1.0.2-1.x86_64
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker18.06.2 will be installed
--> Processing Dependency: docker = 2:18.06.2 for package: nvidia-docker2-2.0.3-1.docker18.06.2.noarch
--> Running transaction check
---> Package libnvidia-container1.x86_64 0:1.0.2-1 will be installed
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker18.06.2 will be installed
--> Processing Dependency: docker = 2:18.06.2 for package: nvidia-docker2-2.0.3-1.docker18.06.2.noarch
--> Finished Dependency Resolution
Error: Package: nvidia-docker2-2.0.3-1.docker18.06.2.noarch (nvidia-docker)
           Requires: docker = 2:18.06.2
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

### 3. Information to [attach](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) (optional if deemed irrelevant)
# docker version
Client:
 Version:           18.06.2-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        6d37f41
 Built:             Sun Feb 10 03:46:03 2019
 OS/Arch:           linux/amd64
 Experimental:      false

**Comment by zhaolupingnick** [comment]
This is because it conflicts with the previous installation of docker-ce, use the following command to check：
 `$ yum install nvidia-docker2-2.0.3-1.docker18.06.2  --skip-broken`

>  You could try using --skip-broken to work around the problem
** Found 2 pre-existing rpmdb problem(s), 'yum check' output follows:
containerd.io-1.2.6-3.3.el7.x86_64 has installed conflicts containerd: containerd.io-1.2.6-3.3.el7.x86_64
containerd.io-1.2.6-3.3.el7.x86_64 has installed conflicts runc: containerd.io-1.2.6-3.3.el7.x86_64


This shows which software caused the conflict, follow up:
 `rpm -e containerd.io`

> error: Failed dependencies:
	containerd.io is needed by (installed) docker-ce-3:19.03.0-2.2.rc2.el7.x86_64

follow up:
`rpm -e  docker-ce-3:19.03.0-2.2.rc2.el7.x86_64`

Run the command again
`yum -y install nvidia-docker2-2.0.3-3.docker18.09.6.ce.noarch`

Success！

**Comment by RenaudWasTaken** [comment]
Happy you solved it! Sorry for the late reply :(
